process/smartContract/process.go

```diff
@@ -456,16 +542,16 @@ func (sc *scProcessor) processVMOutput(
 		return nil, nil, err
 	}
 
-	totalGasConsumed := tx.GasLimit - vmOutput.GasRemaining.Uint64()
+	totalGasConsumed := tx.GasLimit - vmOutput.GasRemaining
 	log.Debug("total gas consumed", "value", totalGasConsumed, "hash", txHash)
 
 	if vmOutput.GasRefund.Uint64() > 0 {
 		log.Debug("total gas refunded", "value", vmOutput.GasRefund.Uint64(), "hash", txHash)
 	}
 
 	totalGasRefund := big.NewInt(0)
-	totalGasRefund = totalGasRefund.Add(vmOutput.GasRefund, vmOutput.GasRemaining)
-	sc.gasHandler.SetGasRefunded(totalGasRefund.Uint64(), txHash)
+	totalGasRefund = totalGasRefund.Add(vmOutput.GasRefund, big.NewInt(0).SetUint64(vmOutput.GasRemaining))
+	sc.gasHandler.SetGasRefunded(vmOutput.GasRemaining, txHash)
```

Comment:
There is a computation above for the totalGasRefund, but you are setting just a part of it. Is it ok?

Reply:
gasHandler wants to know only about the actual consumed gas - the gas consumed for processing. The value gasRefund is only for freeing up memory.

Reply:
ok